lib/new_relic/agent/utilization/ecs.rb

```diff
@@ -0,0 +1,22 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'new_relic/agent/utilization/vendor'
+
+module NewRelic
+  module Agent
+    module Utilization
+      class ECS < Vendor
+        vendor_name 'ecs'
+        endpoint ENV['ECS_CONTAINER_METADATA_URI'] || ''
+        headers 'Metadata' => 'true'
+        keys %w[DockerId]
+
+        def transform_key(key)
+          'ecs' + key
```

Comment:
Does this need to pass back up to the `transform_key` method in the parent class?

Reply:
No, this class is the only place would define any key transformations that the parent would make (but we aren't defining any). I had to overwrite the transform_key method because it needed to work slightly differently overall for ECS